vm-deployment/vm.bicep

```
@@ -0,0 +1,218 @@
+param sshPublicKey string
+
+@description('The size of the virtual machine.')
+param vmSize string = 'Standard_B2s'
+
+@description('The location where resources are deployed.')
+param location string = resourceGroup().location
+
+@description('The static public IP address for the virtual machine.')
+param publicIPAddress string = '4.233.216.162'
+
+@description('The private IP address for the virtual machine.')
+param privateIPAddress string = '10.0.0.4'
+
+@description('The address prefix for the virtual network.')
+param vnetAddressPrefix string = '10.0.0.0/16'
+
+@description('The address prefix for the subnet.')
+param subnetAddressPrefix string = '10.0.0.0/24'
+
+@description('Operating system disk size in Gigabytes.')
+param diskSizeGB int = 30
+
+@description('List of ports to allow for inbound TCP traffic. Example: [22, 80, 443].')
+param allowedTcpPorts array = [22, 80, 443]
+
+@description('Username for the Virtual Machine administrator.')
+@maxLength(64)
+@minLength(1)
+param adminUsername string
+
+
+@description('Operating system disk Stock Keeping Unit name.')
+@allowed([
+  'PremiumV2_LRS'
+  'Premium_LRS'
+  'Premium_ZRS'
+  'StandardSSD_LRS'
+  'StandardSSD_ZRS'
+  'Standard_LRS'
+  'UltraSSD_LRS'
+])
+param diskSku string = 'Premium_LRS'
+
+@description('The priority of the Virtual Machine.')
+@allowed([
+  'Regular'
+  'Low'
+  'Spot'
+])
+param vmPriority string = 'Regular'
+
+
+
+// Networking security group
+resource networkSecurityGroup 'Microsoft.Network/networkSecurityGroups@2023-11-01' = {
+  name: networkSecurityGroupName
+  location: location
+  properties: {
+    securityRules: [for (port, portIndex) in allowedTcpPorts: {
+        name: 'AllowTcp${port}'
+        properties: {
+          protocol: 'TCP'
+          destinationPortRange: '${port}'
+          access: 'Allow'
+          priority: 300 + (portIndex * 20)
+          direction: 'Inbound'
+        }
+      }]
+    ddosSettings: {
+      protectionMode: 'VirtualNetworkInherited'
+    }
+  }
+}
+
+// Public IP address
+resource publicIPAddressResource 'Microsoft.Network/publicIPAddresses@2023-11-01' = {
+  name: publicIPAddressName
+  location: location
+  sku: {
+    name: 'Standard'
+    tier: 'Regional'
+  }
+  properties: {
+    publicIPAddressVersion: 'IPv4'
+    publicIPAllocationMethod: 'Static'
+    ddosSettings: {
+      protectionMode: 'VirtualNetworkInherited'
+    }
+  }
+}
+
+
+// Virtual network
+resource virtualNetwork 'Microsoft.Network/virtualNetworks@2023-11-01' = {
+  name: virtualNetworkName
+  location: location
+  properties: {
+    addressSpace: {
+      addressPrefixes: [
+        vnetAddressPrefix
+      ]
+    }
+    subnets: [
+      {
+        name: 'default'
+        properties: {
+          addressPrefix: subnetAddressPrefix
+          privateEndpointNetworkPolicies: 'Disabled'
+          privateLinkServiceNetworkPolicies: 'Enabled'
+        }
+      }
+    ]
+  }
+}
+
+
+
+// Virtual machine
```

```suggestion
}

// Virtual machine
```
A single newline between two resources.